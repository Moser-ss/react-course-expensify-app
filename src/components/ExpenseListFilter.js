import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize'
import { DateRangePicker } from 'react-dates'
//import 'react-dates/lib/css/_datepicker.css';
import { setTextFilter , sortByDate , sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilter extends React.Component {
  state = {
    calendarFocused: null
  }
  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate)
    this.props.setEndDate(endDate)
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }))
  }

  onTextChange = (event) => {
    this.props.setTextFilter(event.target.value)
  }

  onSortChange = (event) => {
    switch(event.target.value){
      case 'date':
        this.props.sortByDate()
        break;
      case 'amount':
        this.props.sortByAmount()
        break;
    }
  }

  render(){
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input 
              type="text"
              className="text-input"  
              placeholder="Search expenses"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select 
              className="select"
              value={this.props.filters.sortBy} 
              onChange={this.onSortChange}>
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              startDateId="StartDateID"
              endDate={this.props.filters.endDate}
              endDateId="EndtDateID"
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              displayFormat="DD/MM/YYYY"
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
  </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
  }
}
export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilter);