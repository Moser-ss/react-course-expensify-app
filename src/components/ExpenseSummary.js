import React from 'react'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'
import 'numeral/locales.js'
numeral.locale('pt-pt')

export const ExpenseSummary = (props) => {
  const numberExpenses = props.expenses.length
  let totalMessage
  if(numberExpenses > 0){
    const total = numeral(selectExpensesTotal(props.expenses) / 100 ).format('$0,0.00')
    totalMessage = `totalling ${total}`
  }
  return (  <div>
    <p>Viewing {numberExpenses} 
    {numberExpenses === 1 ?' expense': ' expenses'} 
    {totalMessage && ` ${totalMessage}` }</p>
  </div>)
}



const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseSummary)


