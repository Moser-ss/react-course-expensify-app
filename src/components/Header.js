import React from 'react'
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
      <NavLink to="/" activeClassName="is-active" exact={true}>Expense Dashboard</NavLink>
      <NavLink to="/add" activeClassName="is-active">Add Expense</NavLink>
  </header>
);

export default Header;