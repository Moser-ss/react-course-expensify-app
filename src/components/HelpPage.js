import React from 'react'

const HelpPage = (props) => {
  console.log(props);
  return (
    <div>
      This is from my Help component
    </div>
  );
}

export default HelpPage;