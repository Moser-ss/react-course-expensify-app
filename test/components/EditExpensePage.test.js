import React from 'react'
import { shallow } from 'enzyme'
import { EditExpensePage } from '../../src/components/EditExpensePage'
import expenses from '../data/expenses'

let wrapper, editExpense, removeExpense, history 

beforeEach(() => {
  editExpense = jest.fn()
  removeExpense = jest.fn()
  history = { push: jest.fn() }
  wrapper = shallow(<EditExpensePage
    expense={expenses[0]} 
    editExpense={editExpense} 
    removeExpense={removeExpense}
    history={history}
    />)
})

test('should render EditExpensePage', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle editExpense', () => {
  const expenseEdited = {
    description: 'Package of gum',
    note:'Contains 10 Gums',
    amount: 500,
  }
  wrapper.find('ExpenseFrom').prop('onSubmit')(expenseEdited)
  expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenseEdited)
  expect(history.push).toHaveBeenLastCalledWith('/')
})

test('should handle removeExpense', () => {
  wrapper.find('button').simulate('click')
  const id = expenses[0].id
  expect(removeExpense).toHaveBeenLastCalledWith({id})
  expect(history.push).toHaveBeenLastCalledWith('/')
})