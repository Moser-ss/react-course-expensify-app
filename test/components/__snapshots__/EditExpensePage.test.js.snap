// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseFrom
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "qEtAP7D6Ja",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
